projects/timer_fw102/src/main.c

```diff
@@ -0,0 +1,46 @@
+#include <stdbool.h>
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timing_callback(SoftTimerId timer_id, void *context) {
```

Comment:
This is a private function, so it should be `static` as per the [coding standards](https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/9306218/Coding+Standards). `static` gives the function "internal linkage" (as opposed to regular "external linkage") which basically means that the function is local to the file only. Otherwise, if some other file defined a `prv_timing_callback` function, you might have issues.